subsys/bindesc/bindesc_version.c

```diff
@@ -33,6 +33,11 @@ BINDESC_UINT_DEFINE(kernel_version_number, BINDESC_ID_KERNEL_VERSION_NUMBER,
 			KERNEL_VERSION_NUMBER);
 #endif /* IS_ENABLED(CONFIG_BINDESC_KERNEL_VERSION_NUMBER) */
 
+#if IS_ENABLED(CONFIG_BINDESC_KERNEL_BUILD_VERSION)
+BINDESC_STR_DEFINE(kernel_build_version, BINDESC_ID_KERNEL_BUILD_VERSION,
+			STRINGIFY(BUILD_VERSION));
```

Comment:
If I had to guess, I think it would be related to the extra escaping required on the command line... `-D BUILD_VERSION='"..."'`

https://github.com/zephyrproject-rtos/zephyr/blob/36c12482c6ff4349307c93c67e9cd914ed0c6d1c/CMakeLists.txt#L480-L482

Reply:
actually... I was thinking about `cmake -D BUILD_VERSION=a.b.c`, but I think that should be detached enough, and can be wrapped in appropriate quotes on the line above.